lib/datasource.js

```diff
@@ -347,11 +347,17 @@ DataSource.prototype.setup = function(dsName, settings) {
   var dataSource = this;
   var connector;
 
-  // support single settings object
-  if (dsName && typeof dsName === 'object' && !settings) {
-    // setup({name: 'myDataSource', connector: 'memory'})
-    settings = dsName;
-    dsName = undefined;
+  // First argument is an `object`
+  if (dsName && typeof dsName === 'object') {
+    if (settings === undefined) {
+      // setup({name: 'myDataSource', connector: 'memory'})
+      settings = dsName;
+      dsName = undefined;
+    } else {
+      // setup(connector, {name: 'myDataSource', host: 'localhost'})
```

Comment:
Please add this example usage to the jsdoc for `setup` and ideally to jsdoc for `new DataSource()` too.